app/models/manageiq/providers/vmware/infra_manager/inventory/parser/virtual_machine.rb

```diff
@@ -65,7 +65,7 @@ def parse_virtual_machine_summary_runtime(vm_hash, props)
       return if runtime.nil?
 
       vm_hash[:connection_state] = runtime[:connectionState]
-      vm_hash[:host] = persister.hosts.lazy_find(runtime[:host]) if runtime[:host]
+      vm_hash[:host] = persister.hosts.lazy_find(runtime[:host]._ref) if runtime[:host]
```

Comment:
with the nested lazy_find, we should pass InventoryObject or InventoryObjectLazy into lazy_find, doesn't this explose in specs (do we test this line with runtime[:host] not nil?)

Reply:
Yes it is tested, I found this in the assert_specific_vm test that was added see the last two checks for the vm.host

Reply:
hm, what does the ```._ref``` results to?

Reply:
ah, don't mind me, we just expect Host's ems_ref here, so it all looks great :-)

Reply:
Yeah exactly, `runtime[:host]` is an `RbVmomi::VIM::HostSystem` object when I thought it was the host's ref.  The `#_ref` method gets the ems_ref of any `ManagedObject` child class.  https://github.com/vmware/rbvmomi/blob/master/lib/rbvmomi/basic_types.rb#L172-L188